articles/dms/howto-mysql-to-azure-mysql-powershell.md

```diff
@@ -0,0 +1,657 @@
+    LogMessage -Message "Created resource group - $($resourceGroup.ResourceId)."
+}
+else { LogMessage -Message "Resource group $ResourceGroupName exists." }
+```
+
+## Step 4: Create an instance of Azure Database Migration Service
+
+You can create new instance of Azure Database Migration Service by using the [New-AzDataMigrationService](powershell/module/az.datamigration/new-azdatamigrationservice) command. This command expects the following required parameters:
+* *Azure Resource Group name*. You can use [New-AzResourceGroup](/powershell/module/az.resources/new-azresourcegroup) command to create Azure Resource group as previously shown and provide its name as a parameter.
+* *Service name*. String that corresponds to the desired unique service name for Azure Database Migration Service 
+* *Location*. Specifies the location of the service. Specify an Azure data center location, such as West US or Southeast Asia
+* *Sku*. This parameter corresponds to DMS Sku name. The currently supported Sku name are *Standard_1vCore*, *Standard_2vCores*, *Standard_4vCores*, *Premium_4vCores*.
+* *Virtual Subnet Identifier*. You can use [Get-AzVirtualNetworkSubnetConfig](/powershell/module/az.network/get-azvirtualnetworksubnetconfig) command to get the information of a subnet. 
+
+The following script expects that the *myVirtualNetwork* virtual network exists with a subnet named *default* and then creates a Database Migration Service with the name *myDmService* under the resource group created in **Step 3** and in the same region.
+
+```powershell
+# Get a reference to the DMS service - Create if not exists
+[string] $VirtualNetworkName = "myVirtualNetwork"
+[string] $SubnetName = "default"
+[string] $ServiceName = "myDmService"
+
+$dmsServiceResourceId = "/subscriptions/$($global:currentSubscriptionId)/resourceGroups/$ResourceGroupName/providers/Microsoft.DataMigration/services/$ServiceName"
+$dmsService = Get-AzResource -ResourceId $dmsServiceResourceId -ErrorAction SilentlyContinue
+
+# Create Azure DMS service if not existing
+# Possible values for SKU currently are Standard_1vCore,Standard_2vCores,Standard_4vCores,Premium_4vCores
+if (-not($dmsService)) {   
+    $virtualNetwork = Get-AzVirtualNetwork -ResourceGroupName $ResourceGroupName -Name $VirtualNetworkName
+    if (-not ($virtualNetwork)) { throw "ERROR: Virtual Network $VirtualNetworkName does not exists" }
+
+    $subnet = Get-AzVirtualNetworkSubnetConfig -VirtualNetwork $virtualNetwork -Name $SubnetName
+    if (-not ($subnet)) { throw "ERROR: Virtual Network $VirtualNetworkName does not contains Subnet $SubnetName" }
+
+    LogMessage -Message "Creating Azure Data Migration Service $ServiceName..." -IsProcessing $true
+    New-AzDataMigrationService `
+        -ResourceGroupName $ResourceGroupName `
+        -Name $ServiceName `
+        -Location $resourceGroup.Location `
+        -Sku Premium_4vCores `
+        -VirtualSubnetId $Subnet.Id | Out-Null
```

Comment:
If you want to save a reference to the project, just remove the | Out-Null statement.  Then you don't need to get the resource online 225.

Reply:
New-AzDataMigrationService also does not returns ResourceId which is returned by all create cmdlets 
That is the reason I have used Get-AzResource which returns a standard output. Hope that is fine?

Reply:
Done